pkg/cli/clisqlclient/api.go

```diff
@@ -112,8 +112,8 @@ type ServerInfo struct {
 	ClusterID string
 	// Organization is the cluster organization of the remote server.
 	Organization string
-	// VirtualClusterName is the name of the virtual cluster the SQL
-	// client is connected to.
+	// VirtualClusterName is the name of the virtual cluster the SQL client is
+	// connected to. Empty if no application VCs have been defined.
```

Comment:
I suppose the other case in which it might be empty is if we are talking to a serverless cluster where I think we may still have some tenants with no name field.  We've made such tenants very hard to produce in 23.1. I do think this points out another potential behaviour difference because of this change. Namely, in CC I think some people will get VC names in their shell and others won't depending on when their tenant was created.  It's potentially also true that some people will see `tenant-%d` (the old naming scheme) while some will see `cluster-%d` the new naming scheme.  Maybe we can fix this by renaming a bunch of tenants in serverless. 

I don't think this needs to block this PR. Just something we might need to figure out with CC.